### PR TITLE
Added VersionConverter

### LINKED ISSUE #41368: System.Text.Json - JsonSerializer issue when deserializing property of type System.Version
### Description
Property of type System.Version serialized successfully, but on deserialization it returns empty object (in sample below Ver1 = 0.0.0.0)

public class TestClass
{
    public Version Ver1 { get; set; }
}

var ver = new TestClass
{
    Ver1 = new Version(1, 3, 4, 6)
}

string serialized = JsonSerializer.Serialize(ver);    // returns {"Ver1":"Major":1,"Minor":3,"Build":4,"Revision":6,"MajorRevision":0,"MinorRevision":6}}

TestClass deser = JsonSerializer.Deserialize<TestClass>(serialized);    // Ver1 is 0.0

### Configuration
Windows 10 2004 X64
.NET Core 3.1.7

### PR DESCRIPTION
**Pull request by devsko**
Fixes #41368

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        public override Version? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            string? versionString = reader.GetString();
+            if (Version.TryParse(versionString, out Version? result))
+            {
+                return result;
+            }
+
+            ThrowHelper.ThrowJsonException();
```
Comment (steveharter) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs:16]: I think throwing like this is fine (where we have raw JsonException without special text for Version and without inner exception from Parse). So as-is we'll get a fairly nice message like this:
```
System.Text.Json.JsonException : The JSON value could not be converted to System.Version. Path: $ | LineNumber: 0 | BytePositionInLine: 2.
```

**Review by steveharter — APPROVED** [review]

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs (hunk 1 above)
Comment (layomia) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs:16]: Yup we have the same pattern for [Uri](https://github.com/dotnet/runtime/blob/98cba4cec14274fd2abaa88534814eac1e7eb573/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UriConverter.cs#L16) and [KeyValuePair](https://github.com/dotnet/runtime/blob/98cba4cec14274fd2abaa88534814eac1e7eb573/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/KeyValuePairConverter.cs#L81).

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class VersionConverter : JsonConverter<Version>
+    {
+        public override Version? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```
Comment (layomia) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs:8]: We should have matching nullability here.

```suggestion
        public override Version Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

**Review by layomia — CHANGES_REQUESTED** [review]

**Review by layomia — APPROVED** [review]
Thanks.

**Comment by layomia** [comment]
Failing System.Net.Http.Functional.Tests are unrelated.

**Pull request merged**